pkg/controllers/disruption/consolidation_test.go

```diff
@@ -3508,58 +3364,492 @@ var _ = Describe("Consolidation", func() {
-			for i := 0; i < 2; i++ {
-				extraPods = append(extraPods, test.Pod(test.PodOptions{
-					ResourceRequirements: v1.ResourceRequirements{
-						Requests: v1.ResourceList{v1.ResourceCPU: *resource.NewQuantity(1, resource.DecimalSI)},
+			// controller should finish
+			Eventually(finished.Load, 10*time.Second).Should(BeTrue())
+			wg.Wait()
+
+			// should have no nodeClaims deleted from single nodeClaim consolidation
+			Expect(ExpectNodeClaims(ctx, env.Client)).To(HaveLen(numNodes))
+		})
+	})
+	Context("Multi-NodeClaim", func() {
+		var nodeClaims, spotNodeClaims []*v1beta1.NodeClaim
+		var nodes, spotNodes []*v1.Node
+
+		BeforeEach(func() {
+			nodeClaims = []*v1beta1.NodeClaim{}
+			spotNodeClaims = []*v1beta1.NodeClaim{}
+			nodes = []*v1.Node{}
+			spotNodes = []*v1.Node{}
+			nodeClaims, nodes = test.NodeClaimsAndNodes(3, v1beta1.NodeClaim{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
+						v1beta1.NodePoolLabelKey:     nodePool.Name,
+						v1.LabelInstanceTypeStable:   mostExpensiveInstance.Name,
+						v1beta1.CapacityTypeLabelKey: mostExpensiveOffering.CapacityType,
+						v1.LabelTopologyZone:         mostExpensiveOffering.Zone,
 					},
-				}))
-			}
-			ExpectApplied(ctx, env.Client, extraPods[0], extraPods[1])
-			// bind the extra pods to node1 and node 2 to make the consolidation decision invalid
-			// we bind to 2 nodes so we can deterministically expect that node3 is consolidated in
-			// single nodeclaim consolidation
-			ExpectManualBinding(ctx, env.Client, extraPods[0], node1)
-			ExpectManualBinding(ctx, env.Client, extraPods[1], node2)
+				},
+				Status: v1beta1.NodeClaimStatus{
+					Allocatable: map[v1.ResourceName]resource.Quantity{
+						v1.ResourceCPU:  resource.MustParse("32"),
+						v1.ResourcePods: resource.MustParse("100"),
+					},
+				},
+			})
+			spotNodeClaims, spotNodes = test.NodeClaimsAndNodes(3, v1beta1.NodeClaim{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
+						v1beta1.NodePoolLabelKey:     nodePool.Name,
+						v1.LabelInstanceTypeStable:   mostExpensiveSpotInstance.Name,
+						v1beta1.CapacityTypeLabelKey: mostExpensiveSpotOffering.CapacityType,
+						v1.LabelTopologyZone:         mostExpensiveSpotOffering.Zone,
+					},
+				},
+				Status: v1beta1.NodeClaimStatus{
+					Allocatable: map[v1.ResourceName]resource.Quantity{
+						v1.ResourceCPU:  resource.MustParse("32"),
+						v1.ResourcePods: resource.MustParse("100"),
+					},
+				},
+			})
+		})
+		DescribeTable("can merge 3 nodes into 1",
+			func(spotToSpot bool) {
+				nodeClaim1 := lo.Ternary(spotToSpot, spotNodeClaims[0], nodeClaims[0])
+				nodeClaim2 := lo.Ternary(spotToSpot, spotNodeClaims[1], nodeClaims[1])
+				nodeClaim3 := lo.Ternary(spotToSpot, spotNodeClaims[2], nodeClaims[2])
+				node1 := lo.Ternary(spotToSpot, spotNodes[0], nodes[0])
+				node2 := lo.Ternary(spotToSpot, spotNodes[1], nodes[1])
+				node3 := lo.Ternary(spotToSpot, spotNodes[2], nodes[2])
```

Comment:
Any reason why you can't do the following? 

```suggestion
				nodeClaims := lo.Ternary(spotToSpot, spotNodeClaims, nodeClaims)
				nodes := lo.Ternary(spotToSpot, spotNodes, nodes)
```

Reply:
Addressed this test case accordingly.